incubator/binding-mqtt-kafka/src/main/java/io/aklivity/zilla/runtime/binding/mqtt/kafka/internal/stream/MqttKafkaSubscribeFactory.java

```diff
@@ -346,7 +453,14 @@ private void onMqttData(
             assert initialAck <= initialSeq;
 
             doMqttReset(traceId);
-            delegate.doKafkaAbort(traceId, authorization);
+            if (MqttKafkaState.initialOpened(messages.state))
+            {
+                messages.doKafkaAbort(traceId, authorization);
+            }
```

Comment:
Why is this conditional at the caller based on internal state of `messages` stream?

Usually we check internal state inside the method and behave accordingly.
Please take a look at other similar method calls in the onMqtt* method below.

Reply:
It can happen that e.g we get an mqtt abort, but the messages stream is not even open yet, as we still delivering the retained messages.

Reply:
That's a comment about why we need the logic. My feedback was not that the logic is unnecessary, it was about where the logic lives.